rfcs/rfc0012.md

```diff
@@ -0,0 +1,176 @@
+# Configure option for not including taint support
+
+## Preamble
+
+    Author:  Neil Bowers <neilb@cpan.org>
+    Sponsor:
+    ID:      0012
+    Status:  Draft
+
+
+## Abstract
+
+Add a Configure option for deciding whether to include taint support
+in this build of Perl.
+This is essentially a Configure wrapper
+around the existing `-DNO_TAINT_SUPPORT` compile flag.
+The question would be worded as
+
+> "Do you want to build Perl with taint support? [y]",
+
+so the default of "Yes" means that Perl will build as previously.
+
+## Motivation
+
+* Support for taint adds a runtime overhead of roughly 10%.
+* Adding a Configure option makes it easier for people to build a
+  Perl without taint support.
+* Many people don't even know there's an option to build Perl without
+  taint support, or what the benefit is.
+  This will improve awareness.
+* The default of "yes" means that if people aren't paying attention
+  when running Configure, then they'll get a perl similar to the last
+  one they configured, and no code will unexpectedly break.
+* Having this as a Configure option means you'll then be able to
+  query %Config to determine whether your
+  perl was built with taint support.
+
+## Rationale
+
+* Anecdotally, a very high percentage of Perl developers never use
+  the taint features, yet we're all taking the performance hit.
+* We want the default to be that Perl continues to be built with
+  taint support, for backwards compatibility.
+* An easier question would be "Do you want taint support?",
+  but my assumption is that Configure options should default to "No",
+  which is why this is proposing a slightly more clumsy question.
+
+## Specification
+
+### What do we mean by "no Taint support"?
+
+There are currently two compilation options:
+
+1. `SILENT_NO_TAINT_SUPPORT`: no taint support, but you can still call `-T`,
+   it just won't do anything.
+2. `NO_TAINT_SUPPORT`: no taint support, trying to call `-T` is fatal.
+
+I propose that the Configure option sets `NO_TAINT_SUPPORT`.
```

Comment:
I vehemently disagree with this. A [quick CPAN grep](https://grep.metacpan.org/search?size=20&q=%5E%23%5B%5E%5Cn%5D*perl%5B%5E%5Cn%5D-T&qd=&qft=.t) stopped looking for more test files that would be broken after the first 2000.

Taint is a stricture, much like `use strict`. Just like programs shouldn't fail when you remove `use strict`, they shouldn't either when you remove taintedness. Making it fail loudly is a fine option for research purposes, but not something that will result in a usable CPAN in any near future.

Reply:
I don't see this as a show-stopper: just ignore the `-T` from those lines. My bet is that lots of scripts added it from a template or just because "they were used to do so".
Is there any other reason to oppose?

Reply:
> I don't see this as a show-stopper: just ignore the -T from those lines.

That's exactly what `SILENT_NO_TAINT_SUPPORT` does, that was my point.

> My bet is that lots of scripts added it from a template or just because "they were used to do so".

Probably. I have only once used it on purpose in a test (and it's not coincidence it was named `taint.t`).